packages/core/test/acceptance/control_flow_exploration_spec.ts

```diff
@@ -366,28 +366,19 @@ describe('control flow', () => {
          const calls: string[][] = [];
 
          @Component({
-           template: `{#for (item of items); track trackingFn(item, compProp)}{{item}}{/for}`,
+           template: `{#for (item of items); track trackingFn($index, items)}({{item}}){/for}`,
          })
          class TestComponent {
            items = ['one', 'two', 'three'];
-           compProp = 'hello';
 
-           trackingFn(item: string, message: string) {
-             calls.push([item, message]);
-             return item;
+           trackingFn(index: number, items: string[]) {
+             return items[index];
```

Comment:
WDYT about extending this a bit with
```suggestion
           trackingFn(index: number, items: string[]) {
             trackByCalled = true;
             expect(typeof index).toBe('number');
             expect(items).toBe(this.items);
             return items[index];
```

and then verifying that `trackByCalled` is indeed true after rendering, as the current tests would also work if the tracking function is never invoked at all.

Reply:
Or perhaps better yet would be to maintain a `Map<number, string[]>` to track all indices and their `items` value, which should end up containing three items.

Reply:
Have an alternative (but also deduping approach in https://github.com/angular/angular/pull/51980)